src/data/services/LmsApiService.js

```diff
@@ -5,19 +5,36 @@ class LmsApiService {
   static baseUrl = configuration.LMS_BASE_URL;
   static pageSize = 25
 
-  static fetchGradebookData(courseId, searchText, cohort, track) {
-    let gradebookUrl = `${LmsApiService.baseUrl}/api/grades/v1/gradebook/${courseId}/`;
-
-    gradebookUrl += `?page_size=${LmsApiService.pageSize}&`;
+  static fetchGradebookData(courseId, searchText, cohort, track, options = {}) {
+    const queryParams = {};
+    queryParams.page_size = LmsApiService.pageSize;
     if (searchText) {
-      gradebookUrl += `user_contains=${searchText}&`;
+      queryParams.user_contains = searchText;
     }
     if (cohort) {
-      gradebookUrl += `cohort_id=${cohort}&`;
+      queryParams.cohort_id = cohort;
     }
     if (track) {
-      gradebookUrl += `enrollment_mode=${track}`;
+      queryParams.enrollment_mode = track;
+    }
+    if (options.assignmentGradeMax || options.assignmentGradeMin) {
+      if (!options.assignment) {
+        throw new Error('Gradebook LMS API requires assignment to be set to filter by min/max assig. grade');
```

Comment:
Why is this?

Reply:
Can't filter by assignment grade if you haven't told me which assignment grade to look up. Overall course grade is a separate filter.

The assignment grade filter inputs are also disabled in the UI when the assignment isn't chosen, so we won't get into this situation.